### PR TITLE
move docs to root of repo

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by danking** [comment]
don't merge; gonna test docs building on this PR first

File: .github/workflows/python-docs.yml [hunk 1]
```diff
@@ -45,8 +44,8 @@ jobs:
 
           # Only try to commit if there exist changes
           git diff --quiet HEAD || {
-              git commit -m 'Python docs for '$this_sha
+              git commit -m 'Python docs for '$built_sha
               git push origin HEAD:gh-pages-bench
           }
 
-          git checkout 'HEAD@{2}'
+          git checkout $built_sha
```
Comment (danking) [inline .github/workflows/python-docs.yml:51]: Previously if the diff was empty (and thus we did not create a commit), the reflog only had one entry: the checkout. This is more robust: we switch back to the specific SHA that the cleanup actions expect to exist.

**Comment by danking** [comment]
@gatesn updated docs to be a proper workspace package.

File: pyproject.toml [hunk 2]
```diff
@@ -22,18 +22,13 @@ dev-dependencies = [
     "ruff>=0.1.11",
     "pip>=23.3.2",
     "maturin>=1.4.0",
-    "mkdocs>=1.5.3",
-    "mkdocs-material>=9.5.11",
-    "mkdocs-include-markdown-plugin>=6.0.4",
     "mike>=2.0.0",
```
Comment (gatesn) [inline pyproject.toml]: You can remove Mike too

Comment (danking) [reply pyproject.toml]: Removed!

**Review by robert3005 — APPROVED** [review]

**Pull request merged**